src/openrct2/world/TileInspector.cpp

```diff
@@ -30,201 +30,201 @@
-                break;
-            } while (!(tileElement++)->IsLastForTile());
+                    numFoundElements++;
+                    break;
+                } while (!(tileElement++)->IsLastForTile());
+            }
         }
+        return numFoundElements;
     }
-    return numFoundElements;
-}
-
-/**
- * Forcefully removes an element for a given tile
- * @param x The x coordinate of the tile
- * @param y The y coordinate of the tile
- * @param elementIndex The nth element on this tile
- */
-GameActionResultPtr tile_inspector_remove_element_at(const CoordsXY& loc, int16_t elementIndex, bool isExecuting)
-{
-    if (isExecuting)
+
+    /**
+     * Forcefully removes an element for a given tile
+     * @param x The x coordinate of the tile
+     * @param y The y coordinate of the tile
+     * @param elementIndex The nth element on this tile
+     */
+    GameActionResultPtr RemoveElementAt(const CoordsXY& loc, int16_t elementIndex, bool isExecuting)
     {
-        // Forcefully remove the element
-        TileElement* const tileElement = map_get_nth_element_at(loc, elementIndex);
-        if (tileElement == nullptr)
+        if (isExecuting)
         {
-            return std::make_unique<GameActions::Result>(GameActions::Status::Unknown, STR_NONE);
-        }
+            // Forcefully remove the element
+            TileElement* const tileElement = map_get_nth_element_at(loc, elementIndex);
+            if (tileElement == nullptr)
+            {
+                return std::make_unique<GameActions::Result>(GameActions::Status::Unknown, STR_NONE);
+            }
 
-        auto largeScenery = tileElement->AsLargeScenery();
-        if (largeScenery)
-        {
-            // Only delete the banner entry if there are no other parts of the large scenery to delete
-            if (numLargeScenerySequences(loc, largeScenery) == 1)
+            auto largeScenery = tileElement->AsLargeScenery();
+            if (largeScenery)
+            {
+                // Only delete the banner entry if there are no other parts of the large scenery to delete
+                if (numLargeScenerySequences(loc, largeScenery) == 1)
+                {
+                    tileElement->RemoveBannerEntry();
+                }
+            }
+            else
             {
+                // Removes any potential banners from the entry
                 tileElement->RemoveBannerEntry();
             }
-        }
-        else
-        {
-            // Removes any potential banners from the entry
-            tileElement->RemoveBannerEntry();
-        }
 
-        tile_element_remove(tileElement);
-        map_invalidate_tile_full(loc);
+            tile_element_remove(tileElement);
+            map_invalidate_tile_full(loc);
 
-        // Update the window
-        rct_window* const tileInspectorWindow = window_find_by_class(WC_TILE_INSPECTOR);
-        if (tileInspectorWindow != nullptr && loc == windowTileInspectorTile.ToCoordsXY())
-        {
```

Comment:
It looks like that now this check is removed, the code bellow will be called for anyone, even if they have a different tile selected.